KeyVault.Acmebot/Providers/GoDaddyProvider.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+using KeyVault.Acmebot.Options;
+
+using Newtonsoft.Json;
+
+namespace KeyVault.Acmebot.Providers
+{
+    public class GoDaddyProvider : IDnsProvider
+    {
+        private readonly GoDaddyClient _goDaddyClient;
+
+        public GoDaddyProvider(GoDaddyOptions options)
+        {
+            _goDaddyClient = new GoDaddyClient(options.ApiKey, options.Secret);
+        }
+
+        public int PropagationSeconds => 60;
+
+        public async Task CreateTxtRecordAsync(DnsZone zone, string relativeRecordName, IEnumerable<string> values)
+        {
+            var recordName = $"{relativeRecordName}";
+
+            foreach (var value in values)
+            {
+                await _goDaddyClient.CreateTxtRecordAsync(zone.Name, recordName, value);
+            }
+        }
+
+        public Task DeleteTxtRecordAsync(DnsZone zone, string relativeRecordName)
+        {
+            return Task.CompletedTask;

```

Comment:
Need implementation. Upsert is not possible due to its requirements, so it is necessary to implement creation and deletion separately.

Reply:
Not resolved.